compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/diagnostic/BLangDiagnosticLog.java

```diff
@@ -289,4 +301,53 @@ private void storeDiagnosticInModule(PackageID pkgId, Diagnostic diagnostic) {
         BLangPackage pkgNode = this.packageCache.get(pkgId);
         pkgNode.addDiagnostic(diagnostic);
     }
+
+    private boolean hasOtherTypeInArgs(Object[] args) {
+        BType constraint;
+        for (Object arg : args) {
+            if (arg instanceof BType) {
+                switch (((BType) arg).getKind()) {
+                    case OTHER:
+                        return true;
+                    case TUPLE:
+                        List<BTupleMember> members = ((BTupleType) arg).getMembers();
+                        for (BTupleMember bTupleMember : members) {
+                            if (bTupleMember.type.getKind() == TypeKind.OTHER) {
+                                return true;
+                            }
+                        }
+                        break;
+                    case UNION:
+                        LinkedHashSet<BType> memberTypes = ((BUnionType) arg).getMemberTypes();
+                        for (BType memberType : memberTypes) {
+                            if (memberType.getKind() == TypeKind.OTHER) {
+                                return true;
+                            }
+                        }
+                        break;
+                    case TABLE:
+                        constraint = ((BTableType) arg).getConstraint();
+                        if (constraint.getKind() == TypeKind.OTHER) {
+                            return true;
+                        }
+                        break;
+                    case MAP:
+                        constraint = ((BMapType) arg).getConstraint();
+                        if (constraint.getKind() == TypeKind.OTHER) {
+                            return true;
+                        }
+                        break;
+                    case TYPEDESC:
```

Comment:
Don't we have to check `xml`, `array`, other types?

Reply:
Updated